firmware/boot/bootloader.c

```diff
@@ -223,12 +206,19 @@ _Noreturn void bootloader_runInterfaceTask(void)
                 .std_id = APP_VALIDITY_ID,
                 .dlc    = 1,
             };
-            reply.data[0] = (uint8_t)verifyAppCodeChecksum();
+            uint8_t valid_app = (uint8_t)verifyAppCodeChecksum();
+            reply.data[0]     = valid_app;
             io_can_pushTxMsgToQueue(&reply);
 
             // Verify command doubles as exit programming state command.
             update_in_progress = false;
         }
+        else if (command.std_id == GO_TO_APP && !update_in_progress)
```

Comment:
So in this case the only way we got from boot to app code is once we successfully flash. I am wondering if we would a kind of a state indicator to indicate we have not exited boot loader

Reply:
In that case we would have unsuccessfully flashed, which is shown in the terminal, so not sure if this would be needed. We could also check CAN messages and see that we are still in boot code.